app/javascript/elm/tests/MainTests.elm

```diff
@@ -849,23 +864,45 @@ updateTests =
                         }
 
                     newSessions =
-                        Encode.list identity
-                            [ Encode.object
-                                [ ( "title", Encode.string "title" )
-                                , ( "id", Encode.int <| id + 1 )
-                                , ( "startTime", Encode.int 0 )
-                                , ( "endTime", Encode.int 0 )
-                                , ( "username", Encode.string "username" )
-                                , ( "shortTypes", Encode.list identity [] )
-                                , ( "average", Encode.null )
-                                ]
+                        Encode.object
+                            [ ( "fetched"
+                              , Encode.list identity
+                                    [ Encode.object
+                                        [ ( "title", Encode.string "title" )
+                                        , ( "id", Encode.int <| id + 1 )
+                                        , ( "startTime", Encode.int 0 )
+                                        , ( "endTime", Encode.int 0 )
+                                        , ( "username", Encode.string "username" )
+                                        , ( "shortTypes", Encode.list identity [] )
+                                        , ( "average", Encode.null )
+                                        ]
+                                    ]
+                              )
+                            , ( "availableSessionsCount", Encode.int 1 )
                             ]
                 in
                 model
                     |> update (UpdateSessions newSessions)
                     |> Tuple.first
                     |> .sessions
                     |> Expect.equal [ newSession ]
+        , fuzz int "UpdateSessions decodes the encoded value and updates model.availableSessionsCount" <|
+            \count ->
+                let
+                    model =
+                        { defaultModel | availableSessionsCount = 0, selectedSession = NotAsked }
```

Comment:
Is `selectedSession` needed?

Reply:
Not really, cause it's in the default model anyway. But it's in the similar test case above (line 854) and I think you wrote it, so I assumed you had a reason for that 😄 
I'll just deleted them both if it's not needed anymore.